Glusterfs 3.12.3

nh2 opened the pull request:
###### Motivation for this change

Bugfix pointreleases

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
- [X] Tested on my production cluster

---

FYI @bachp

nh2 committed:
glusterfs: 3.12.1 -> 3.12.2

nh2 committed:
glusterfs: 3.12.2 -> 3.12.3

Ma27 commented:
is it really necessary to include a commit which does a version bump to `3.12.2` if the version will be changed to `3.12.3` in the same branch?

nh2 commented:
> is it really necessary to include a commit which does a version bump to 3.12.2 if the version will be changed to 3.12.3 in the same branch?

@Ma27 No, I am simply upstreaming the commits from my nixpkgs in the order that I've tested them in my production cluster.

I first bumped to `3.12.2`, updated and started testing my production cluster, but before I could upstream it, `3.12.3` got released and so I updated and tested it in another commit.

I could squash the two, but then they would no longer reflect the order in which I tested the upgrades, and it would create extra work for me (e.g. when it's merged to 17.09, I can no longer fast-forward my fork, because the 1-commit-way and the 2-commit-way will conflict) with no perceivable benefits.

domenkozar commented:
@GrahamcOfBorg build glusterfs

GrahamcOfBorg reviewed (commented):
Failure for system: x86_64-darwin

```
error: Package ‘glusterfs-3.12.3’ in /tmp/nix-ofborg/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/31782/pkgs/tools/filesystems/glusterfs/default.nix:186 is not supported on ‘x86_64-darwin’, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

```

GrahamcOfBorg reviewed (commented):
Success for system: x86_64-linux

```
optional arguments:
  -h, --help            show this help message and exit

usage: .eventsdash.py-wrapped [-h] [--port PORT] [--debug]

optional arguments:
  -h, --help   show this help message and exit
  --port PORT  Port
  --debug      Run Server in debug mode
/nix/store/x0in1r0r3j2vjg56w1h4l06xc4b7fzxh-glusterfs-3.12.3
```
